corehq/motech/repeaters/tasks.py

```diff
@@ -41,6 +42,20 @@
 logging = get_task_logger(__name__)
 
 
+@periodic_task(
+    run_every=crontab(day_of_month=27),
+    queue=settings.CELERY_PERIODIC_QUEUE,
+)
+def clean_logs():
+    """
+    Drop MOTECH logs older than 90 days.
+
+    Runs on the 27th of every month.
+    """
+    ninety_days_ago = datetime.now() - timedelta(days=90)
+    RequestLog.objects.filter(timestamp__lt=ninety_days_ago).delete()
```

Comment:
any reason to break this up into chunks, or is it fine to do monolithically?

Reply:
Good question. If I understand correctly, this results in a single delete query with a where clause, and that the benefit of using one query is that Postgres can optimize it however it thinks is best.

@millerdev or @snopoke might know more about how Postgres would do this, and whether it would be better for us to select chunks, to delete piecemeal, and how best to select the chunks.

Reply:
Depends on the volume I guess. How many logs do we generate per day?

For large volumes you can partition it by day / week and then just drop the oldest table like we do for synclogs:

https://github.com/dimagi/commcare-hq/blob/092775d9f83944f77b9ae34fdaebd61f1e2c16ae/corehq/ex-submodules/casexml/apps/phone/tasks.py#L93-L109

Reply:
We are averaging about 3,000 logs per day.

Reply:
that seems tractable for now though in the long run it would be nice to have this partitioned

Reply:
Cool. Yeah, I think partitioning is a great idea. I'll add it to my backlog.